### PR TITLE
Added Depends_on column to manifest file (when FULL_MANIFEST=TRUE)

### PR DESCRIPTION
**Pull request by xross**
With this change any release tools do not have to do any parsing of cmakelist files to build up a complete dependency tree for checking. 

I think it's also a useful addition for anyone wanting a quick inspection of how the dependencies are built up. 

~There is a bit of C&P when I possibly should have used a function...~  Now uses macro.

### CODE REVIEW

**Review by danielpieczko — DISMISSED** [review]

**Comment by xross** [comment]
@TDW89 please note this change

File: doc/dependency_management.rst [hunk 1]
```diff
@@ -55,8 +55,10 @@ The columns in the manifest file are:
 - Branch/tag: the currently checked out branch or tag. A tag takes precedence, so if the head of a branch is explicitly
   checked out, but that changeset has also been tagged, then the tag will be reported here.
 - Changeset: the git commit hash identifying the current changeset checked out in the repository
-- Dependency_Requirement: (hidden) the requirement for the repository from the dependent modules list variable. This can
+- Dependency_requirement: (hidden) the requirement for the repository from the dependent modules list variable. This can
   differ from the current changeset if the developer has manually checked out a different version of that component. This
   column is only displayed if CMake is run with the option ``-D FULL_MANIFEST=TRUE``.
+- Depends_on: (hidden) a list of the libraries the dependency is dependent on. This column is only displayed if CMake i
```
Comment (danielpieczko) [inline doc/dependency_management.rst:61]: typo: single 'i' at the end of the line

File: xcommon.cmake [hunk 2]
```diff
@@ -760,9 +777,25 @@ function(XMOS_REGISTER_DEPS)
...
+                foreach(DEP_MODULE ${LIB_DEPENDENT_MODULES})
+                    parse_dep_string(${DEP_MODULE} DEP_REPO DEP_VERSION DEP_NAME)
+                    math(EXPR DEP_COUNT "${DEP_COUNT} - 1")
+                    set(manifest_str "${manifest_str} ${DEP_NAME}(${DEP_VERSION})")
+                    if(NOT DEP_COUNT EQUAL 0)
+                        set(manifest_str "${manifest_str},")
+                    endif()
+                endforeach()
+            endif()
+            file(APPEND ${MANIFEST_OUT} "${manifest_str}\n")
```
Comment (danielpieczko) [inline xcommon.cmake:795]: Since this chunk is identical to lines 569-582, this would be better in a macro().

**Review by danielpieczko — APPROVED** [review]
Functionally looks good, but I think the duplicated code should be in a macro. Otherwise this is good.

**Review by danielpieczko — COMMENTED** [review]
Code changes look good, but one test is failing - hopefully you just need to revert [this commit](https://github.com/xmos/xcommon_cmake/pull/138/commits/f0d466a8ebd5bd22dc1c8af90a27324d58924de5) from when I added the column that has now been removed.

**Comment by xross** [comment]
I've patched up the test

**Review by danielpieczko — APPROVED** [review]
Nice

**Pull request merged**